### ISSUE TITLE
Build fails when using Realm and CocoaPods

### ISSUE THREAD

**Issue by TiagoVeloso** [facebook/infer#232]
I am trying to run Infer on my project that includes a bunch of Pods. One of them is Realm.

Running xcodebuild without infer the project builds without any issue.
Running with infer it fails with the following message:

```sh
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/tveloso/Library/Developer/Xcode/DerivedData/ABC-dbvlmexxbrkpqubvzgqzuzdlxptp/Build/Intermediates/Pods.build/Debug-iphoneos/Pods-ABC-Realm.build/Objects-normal/armv7/external_commit_helper.o Realm/Realm/ObjectStore/impl/apple/external_commit_helper.cpp normal armv7 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
None
```

I am using the latest xcodebuild and infer.

**Comment by martinoluca** [comment]
Hello @TiagoVeloso,
I will investigate this issue and give you updates!
In the meanwhile, could you please send me a paste of the real error? What you provided is just a summary of the errors that happened during the build. The real message should be a bit above that line.

Thanks!

**Comment by TiagoVeloso** [comment]
This was the first occurrence I found. But there are a couple of errors like this one:

```sh
While building module 'CoreFoundation' imported from /Users/tveloso/Developer/abc/Pods/Realm/include/realm/external_commit_helper.hpp:22:
While building module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:11:
While building module 'std' imported from /Users/tveloso/Developer/infer-osx-v0.5.0/facebook-clang-plugins/clang/bin/../include/c++/v1/math.h:309:
In file included from <module-includes>:1:
In file included from /Users/tveloso/Developer/infer-osx-v0.5.0/facebook-clang-plugins/clang/bin/../include/c++/v1/algorithm:624:
In file included from /Users/tveloso/Developer/infer-osx-v0.5.0/facebook-clang-plugins/clang/bin/../include/c++/v1/initializer_list:47:
In file included from /Users/tveloso/Developer/infer-osx-v0.5.0/facebook-clang-plugins/clang/bin/../include/c++/v1/cstddef:43:
/Users/tveloso/Developer/infer-osx-v0.5.0/facebook-clang-plugins/clang/bin/../include/c++/v1/stddef.h:46:15: fatal error: cyclic dependency in module 'Darwin': Darwin -> std -> Darwin
#include_next <stddef.h>
              ^
```

if you need more of my build out type let me know

**Comment by martinoluca** [comment]
Could you please provide the command you used to build?

Thanks.

**Comment by TiagoVeloso** [comment]
Something like this:

```sh
infer -- xcodebuild -workspace ABC.xcworkspace -scheme Debug -sdk iphonesimulator -destination 'platform=iOS Simulator,OS=9.2,name=iPhone 6s' clean build
```

**Comment by martinoluca** [comment]
The error paste you provided is not enough, please make sure you see the `error: ` message of clang.
Does it compile if you just run the build command?
```
xcodebuild -workspace ABC.xcworkspace -scheme Debug -sdk iphonesimulator -destination 'platform=iOS Simulator,OS=9.2,name=iPhone 6s'
```

**Comment by TiagoVeloso** [comment]
Yes without infer it builds without an issue.

In attachment is the complete log file of the build process

[out.txt](https://github.com/facebook/infer/files/81361/out.txt)

**Comment by martinoluca** [comment]
Ok seems like this is a [known problem in clang](http://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20151012/140965.html) and for now what I recommend to do is to append 
```
CLANG_ENABLE_MODULES=NO
```
to your build command, so for example in your case it will become:
```
infer -- xcodebuild -workspace ABC.xcworkspace -scheme Debug -sdk iphonesimulator -destination 'platform=iOS Simulator,OS=9.2,name=iPhone 6s' CLANG_ENABLE_MODULES=NO
```

Please let me know if that helps!

**Comment by joshluongo** [comment]
Is there any workaround besides using `CLANG_ENABLE_MODULES=NO`. The reason why I can't use is it is my project requires modules.

**Comment by TiagoVeloso** [comment]
I am in the same situation as @joshluongo I wasn't able to compile with that flag as well.

**Comment by joshluongo** [comment]
I found a simple solution to this problem if your app uses modules, Simply switch off `Enable Modules (C and Objective-C)` in the target for 'Realm'. http://imgur.com/PSfw7ks

If you do that then you don't have to add `CLANG_ENABLE_MODULES=NO` to your command.

**Comment by jvillard** [comment]
Should be fixed in latest master (see #335).
